utils/dataset_manifest/core.py

```diff
@@ -487,24 +502,21 @@ def _write_core_part(self, file, _tqdm):
                 }, separators=(',', ':'))
                 file.write(f"{json_item}\n")
 
-    # pylint: disable=arguments-differ
-    @_set_index
-    def create(self, _tqdm=None):
+    def create(self, *, _tqdm=None): # pylint: disable=arguments-differ
         """ Creating and saving a manifest file """
         if not len(self._reader):
-            with NamedTemporaryFile(mode='w', delete=False)as tmp_file:
-                self._write_core_part(tmp_file, _tqdm)
-            temp = tmp_file.name
+            tmp_file = StringIO()
```

Comment:
Why do we need a `tmp_file` at all? Why not just write directly to `manifest_file`?

Reply:
I'm not sure. Probable, the reason is to avoid creating the file and the removing it in case of an error.

Reply:
I think I figured it out. It's because `VideoStreamReader.__len__` can return `None` until you iterate over it once, and `_write_core_part` needs to know the final length.